From 2086c637711bf4086e1c3faea2c7f8cf6a19b7b7 Mon Sep 17 00:00:00 2001
From: Pooja Prajod <a0132412@ti.com>
Date: Thu, 2 Apr 2015 16:50:17 +0530
Subject: [gst-ducati] Solve memory leak incase of external bufferpool

Gst ducati plugin accepts external bufferpool from peer. This
bufferpool is not freed after use which leads to memleak.

Signed-off-by: Pooja Prajod <a0132412@ti.com>
---
 src/gstducatividdec.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/gstducatividdec.c b/src/gstducatividdec.c
index 9735ff1..84bbf79 100644
--- a/src/gstducatividdec.c
+++ b/src/gstducatividdec.c
@@ -1696,6 +1696,12 @@ gst_ducati_viddec_finalize (GObject * obj)
   GstDucatiVidDec *self = GST_DUCATIVIDDEC (obj);
 
   codec_delete (self);
+
+  if (self->externalpool) {
+    gst_object_unref (self->externalpool);
+    self->externalpool = NULL;
+  }
+
   engine_close (self);
 
   /* Will unref the remaining buffers if needed */
-- 
1.7.9.5

